### PR TITLE
Editorial: Distribution of colorspaceData within instance resources and reviewables

### PR DESCRIPTION
**Pull request by jakubjezek001**
## Changelog Description
Editorial supports distribution of colorspace data within editorial otio media resources and reviewable clips.

## Additional info
Within the scope of https://github.com/ynput/ayon-hiero/issues/22
This needs to be tested together with https://github.com/ynput/ayon-hiero/pull/23

## Testing notes:
1. all should work as it was, but colorspace in editorial is now distributed

### CODE REVIEW

File: client/ayon_core/plugins/publish/extract_color_transcode.py [hunk 1]
```diff
@@ -154,12 +153,15 @@ def process(self, instance):
...
                     files_to_convert)
+                self.log.debug("Files to convert: {}".format(files_to_convert))
                 for file_name in files_to_convert:
+                    self.log.debug("Transcoding file: `{}`".format(file_name))
                     input_path = os.path.join(original_staging_dir,
                                               file_name)
                     output_path = self._get_output_file_path(input_path,
                                                              new_staging_dir,
                                                              output_extension)
+                    self.log.debug("Ynput path: `{}`".format(input_path))
```
Comment (BigRoy) [inline client/ayon_core/plugins/publish/extract_color_transcode.py:164]: This seems very verbose with all these added logs. Intentional? or just for debugging?

```suggestion
                    self.log.debug("Input path: `{}`".format(input_path))
```

Comment (jakubjezek001) [reply client/ayon_core/plugins/publish/extract_color_transcode.py:164]: Huh, and I was assuming this was removed, thanks!

File: client/ayon_core/plugins/publish/collect_otio_review.py [hunk 2]
```diff
@@ -95,9 +95,46 @@ def process(self, instance):
...
 
+            # get colorspace from metadata if available
+            if len(otio_review_clips) >= 1 and any(
+                # lets make sure any clip with media reference is found
+                (
+                    clip
+                    for clip in otio_review_clips
+                    if isinstance(clip, otio.schema.Clip)
+                    and clip.media_reference
+                )
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/collect_otio_review.py:110]: You don't need t ocreate tuple first, you can just use iteration as is.
```suggestion
                clip
                for clip in otio_review_clips
                if (
                    isinstance(clip, otio.schema.Clip)
                    and clip.media_reference
                )
```

File: client/ayon_core/plugins/publish/collect_otio_review.py [hunk 3]
```diff
@@ -95,9 +95,46 @@ def process(self, instance):
             instance.data["label"] = label + " (review)"
             instance.data["families"] += ["review", "ftrack"]
             instance.data["otioReviewClips"] = otio_review_clips
+
             self.log.info(
                 "Creating review track: {}".format(otio_review_clips))
 
+            # get colorspace from metadata if available
+            if len(otio_review_clips) >= 1 and any(
+                # lets make sure any clip with media reference is found
```
Comment (iLLiCiTiT) [inline client/ayon_core/plugins/publish/collect_otio_review.py:104]:
```suggestion
                # make sure any clip with media reference is found
```

**Review by robin-ynput — APPROVED** [review]
Tested locally along hiero changes. Confirms it works and colorspace is applied properly on reviewables.

**Pull request merged**